registry/descriptor.go

```diff
@@ -0,0 +1,40 @@
+package registry

```

Comment:
As discussed, haven't reviewed code in registry package as it going to removed in next iteration.